### DISCUSSION TITLE
What is the reason to have @tanstack/react-query

### DISCUSSION THREAD

**Discussion by sergical** [family/connectkit#357] [Q&A] [answered]
From the documentation: https://docs.family.co/connectkit/getting-started

However, looking at the layout:

```react
export const Web3Provider = ({ children }) => {
  return (
    <WagmiProvider config={config}>
      <QueryClientProvider client={queryClient}>
        <ConnectKitProvider>{children}</ConnectKitProvider>
      </QueryClientProvider>
    </WagmiProvider>
  );
};
```

It's wrapped outside of ConnectKitProvider so it doesn't seem to be necessary. I think the consumers should be able to use anything they want (swr, redux, etc)

**Comment by lochie** [comment, marked as answer]
`@tanstack/react-query` is a setup requirement of wagmi. You can read more about that here:
https://wagmi.sh/react/getting-started#setup-tanstack-query
